### PR TITLE
Add polar decompositions

### PR DESCRIPTION
**Pull request by RalphAS**
This change imports updated versions of the polar decomposition routines
originally implemented by Weijian Zhang (https://github.com/weijianzhang/PolarFact.jl).

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,11 +1,12 @@
 name = "MatrixFactorizations"
 uuid = "a3b82374-2e81-5b9e-98ce-41277c0e4c87"
-version = "0.6.1"
+version = "0.6.2-DEV"
```
Comment (dlfivefifty) [inline Project.toml:3]: Don't bother with -dev as I'll tag as soon as merged.

It should be `"0.7" as you export new names

File: test/test_polar.jl [hunk 2]
```diff
@@ -0,0 +1,180 @@
+@testset "Polar Decomposition" begin
+
+Random.seed!(1103)
+
+for T in [Float64, Complex{Float64}]
+    @testset "Newton $T" begin
+        for n in [1,3,10]
+
```
Comment (dlfivefifty) [inline test/test_polar.jl:8]: Too many newlines here

**Review by dlfivefifty — CHANGES_REQUESTED** [review]
Looks good! Just some small changes.

In the future we may wish to consider splitting out the iterative solves to another package, e.g.,  IterativeFactorizations.jl

File: test/test_polar.jl (hunk 2 above)
Comment (RalphAS) [reply test/test_polar.jl]: Ok, while I'm at it I'll make the test script more like the others

**Comment by dlfivefifty** [comment]
Can you get the coverage up?

**Comment by dlfivefifty** [comment]
Why is OS X allowed to fail?

**Comment by RalphAS** [comment]
I suppressed OSX because I encountered infrastructure failures like [this one](https://travis-ci.org/github/JuliaMatrices/MatrixFactorizations.jl/jobs/743832159) unrelated to the package; the alternative is to ask you to manually restart the checks until they find a lucky instance.

**Review by dlfivefifty — APPROVED** [review]

**Pull request merged**